lib/capabilities.go

```diff
@@ -78,3 +91,31 @@ func CapAwayNotify(caps *CapManager) {
 		},
 	)
 }
+
+func CapServerTime(caps *CapManager) {
+	name := "server-time"
+	caps.Supported[name] = ""
+
+	caps.FnsInitListener[name] = func(listener *Listener) {
+		listener.TagsEnabled = true
+	}
+
+	caps.FnsMessageToClient = append(
+		caps.FnsMessageToClient,
+		func(listener *Listener, message *ircmsg.IrcMessage) bool {
+			if !listener.IsCapEnabled(name) {
+				return false
+			}
+
+			_, exists := message.Tags["time"]
+			if !exists {
+				message.Tags["time"] = ircmsg.TagValue{
```

Comment:
The library does it this way to differentiate between the tags `blah1=test`, `blah2=`, and `blah3`. In the first two cases, the values exist (for `blah2` the value's an empty string). In `blah3` the value doesn't exist at all.

I see it the same way when handling ISUPPORT tokens, easier to just have the differentiation in there from the get-go than to have to hack it in later. Especially useful when we're implementing client-to-client tags so we don't accidentally change the meaning/representation of those. For example, if someone sends us `@+test1=`, we should output it to our connected clients as `@+test1=` as well.